File: Guardian/src/app/dto/RegisterUserQuestionnaireAndCompanyDto.ts
import { RequestRegisterNewUserDto } from "./RequestRegisterNewUserDto ";
import { companyDto } from "./companyDto";
import { SaveQuestionDto } from "./SaveQuestionDto";


export interface RegisterUserQuestionnaireAndCompanyDto {
  requestRegisterNewUserDto: RequestRegisterNewUserDto;
  companyDto: companyDto;
 requestSaveQuestionnaireDto: SaveQuestionDto[];
  
  
}

File: UsaloYa.Dto/RegisterUserQuestionnaireAndCompanyDto.cs
using System.ComponentModel.DataAnnotations;

namespace UsaloYa.Dto
{
    public class RegisterUserQuestionnaireAndCompanyDto
    {
        [Required]
        public RequestRegisterNewUserDto RequestRegisterNewUserDto { get; set; }

        [Required]
        public CompanyDto CompanyDto { get; set; }
        [Required]
        public List<RequestSaveQuestionnaireDto> RequestSaveQuestionnaireDto { get; set; }

    }
}
